### DISCUSSION TITLE
Mutation isLoading and query invalidation

### DISCUSSION THREAD

**Discussion by olup** [TanStack/query#715] [General]
Returning queryCache.invalidateQueries to the onSuccess mutation call back, I was hoping that the isLoading value exported by the hook would stay true until the onSuccess call back promise has landed ( like in v1 with status and refetchQueries). But it does not seem to be the case. The mutation itself resolves properly after the call back, but isLoading  - if i am not mistaken - does not seem to. Is there another way I should do this ?

**Comment by ayush987goyal** [comment]
On the same note, what is the best place, post a successful mutation, to await the `queryCache.invalidateQueries` and still keep the `loading` state of the mutation?

**Comment by alinjie** [comment]
@olup Did you ever manage to find a way to achieve this?
